src/azul/chalice.py

```diff
@@ -504,34 +498,109 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
-    def threshold(self, *, errors: int, throttles: int):
-        def wrapper(f):
+    @attr.define(frozen=True, kw_only=True)
+    class _lambda_config:
```

Comment:
Hannes mentioned he didn't like my original idea for this class's name. Maybe this would be better?
```suggestion
    class _lambda_decorator:
```